app/src/main/java/org/wikipedia/dataclient/WikiSite.java

```diff
@@ -96,6 +96,9 @@ public WikiSite(@NonNull Uri uri) {
         if (languageCode.equals(AppLanguageLookUpTable.CHINESE_LANGUAGE_CODE)) {
             languageCode = LanguageUtil.getFirstSelectedChineseVariant();
         }
+
+        // TODO: should use the default language code on the authority to prevent incorrect url such as zh-tw.wikipedia.org.
```

Comment:
This can be fixed in #1695, or maybe we can ignore this because it is an edge case.